### DISCUSSION TITLE
How to generate PDF, JSON or PHP static files at build time like with Astro Endpoints

### DISCUSSION THREAD

**Discussion by PlanetIrata** [gatsbyjs/gatsby#37468] [Help]
Hi, 

Is it possible with Gatsby to generate any type of static file (not only HTML, CSS of JS files) at build time ? For instance generate a JSON feed, a sitemap, or even a PHP file to replace inside it some dynamic values like the url of a form post?

This is possible in [Astro with Endpoints](https://docs.astro.build/en/core-concepts/endpoints/), is there any equivalent feature in Gatsby?

Thanks

**Comment by LekoArts** [comment]
You can use the [`onPostBuild` API](https://www.gatsbyjs.com/docs/reference/config-files/gatsby-node/#onPostBuild) (since it has access to the GraphQL data layer), e.g. like the sitemap plugin: https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-sitemap
